instrumentation/graphql-java/graphql-java-20.0/library/src/main/java/io/opentelemetry/instrumentation/graphql/v20_0/OpenTelemetryInstrumentation.java

```diff
@@ -34,23 +77,157 @@ public InstrumentationState createState(InstrumentationCreateStateParameters par
   @Override
   public InstrumentationContext<ExecutionResult> beginExecution(
       InstrumentationExecutionParameters parameters, InstrumentationState rawState) {
+
     OpenTelemetryInstrumentationState state = ofState(rawState);
-    return helper.beginExecution(state);
+    Context parentContext = Context.current();
+
+    if (!executionInstrumenter.shouldStart(parentContext, state)) {
+      return SimpleInstrumentationContext.noOp();
+    }
+
+    Context context = executionInstrumenter.start(parentContext, state);
+    state.setContext(context);
+
+    return SimpleInstrumentationContext.whenCompleted(
+        (result, throwable) -> {
+          Span span = Span.fromContext(context);
+
+          // TODO: Needs revisions. Exceptions should be recording by using 'span.recordException'
```

Comment:
Shall I remove this comment of leave it in place?

The reason I added this comment in was because `span.addEvent("exception", attributes.build());` did not feel right to me and I didn't know what to do here. The hardcoded event name `exception` felt a bit semantically misleading to me.

Also do you think the span status should be marked as `ERROR` if GraphQL errors exist?

Reply:
> Shall I remove this comment of leave it in place?

I'd remove it because `recordException` can't be used.

> The reason I added this comment in was because span.addEvent("exception", attributes.build());

Do you have any suggestions on how to improve this?

> Also do you think the span status should be marked as ERROR if GraphQL errors exist?

Isn't it already done in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c92955fa2fb8beae9cf599bb164dd9d9cbd15805/instrumentation/graphql-java/graphql-java-common/library/src/main/java/io/opentelemetry/instrumentation/graphql/internal/OpenTelemetryInstrumentationHelper.java#L67

Reply:
I've removed the comment.

I don't have any suggestions on how to improve this so it might be best to leave it the way it is for now.